src/precompiles/modexp.rs

```diff
@@ -3,87 +3,46 @@ use crate::precompiles::{
 };
 use crate::prelude::{PhantomData, Vec, U256};
 use evm::{Context, ExitError};
-use num::BigUint;
+use num::{BigUint, Integer};
 
 pub(super) const ADDRESS: [u8; 20] = super::make_address(0, 5);
 
 pub(super) struct ModExp<HF: HardFork>(PhantomData<HF>);
 
-impl ModExp<Byzantium> {
-    fn adj_exp_len(exp_len: U256, base_len: U256, bytes: &[u8]) -> U256 {
-        let mut exp32_bytes = Vec::with_capacity(32);
-        for i in 0..32 {
+impl<HF: HardFork> ModExp<HF> {
+    fn calc_iter_count(exp_len: u64, base_len: u64, bytes: &[u8]) -> u64 {
+        let mut exp_bytes = Vec::with_capacity(32);
+        for i in 0..exp_len {
             if U256::from(96) + base_len + U256::from(1) >= U256::from(bytes.len()) {
```

Comment:
Now that `base_len` is a `u64` probably don't need to convert all the other stuff in to `U56`

Reply:
Irrelevant now with the changes to this as commented by https://github.com/aurora-is-near/aurora-engine/pull/138/commits/1f49da17bb2c6e660e191e85e157a8794b04111a.